src/tiled/invertyaxishelper.h

```diff
@@ -24,32 +24,41 @@
 #include "preferences.h"
 #include "mapdocumentactionhandler.h"
 
-class InvertYAxisHelper{
-    public:
-        InvertYAxisHelper() {}
-        ~InvertYAxisHelper() {}
-        
-        // Inverts Y coordinate in grid
-        float getY(float y) const
+namespace Tiled{
+    namespace Internal {
+
+        class InvertYAxisHelper
         {
-            // Check if Invert Y Axis is set
-            if(Tiled::Internal::Preferences::instance()->invertYAxis())
+        public:
+            // Constructors
+            InvertYAxisHelper()  {}
+            InvertYAxisHelper(Map* m) : target(m) {}
+                
+            // Inverts Y coordinate in grid
+            qreal tileY(qreal y) const
             {
-                return Tiled::Internal::MapDocumentActionHandler::instance()->mapDocument()->map()->height() - 1 - y;
+                // Check if Invert Y Axis is set
+                if(Preferences::instance()->invertYAxis())
+                {
```

Comment:
Coding style: please put the `{` on the same line as the `if` (and put a space after `if`):
```c++
if (Preferences::instance()->invertYAxis()) {
```